### DISCUSSION TITLE
SessionManagerPlugin is not found while session-manager-plugin --version is successful

### DISCUSSION THREAD

**Discussion by matthieudesprez** [aws/aws-cli#8707] [Q&A] [answered]
Hello,

I'm trying to automate a few aws cli commands through a raycast extension, but when running commands through nodejs execSync I get a weird behaviour.

For instance, if I try to run the following:

```typescript
const commands = [
    `session-manager-plugin --version`,
    `which session-manager-plugin`,
    `which aws`,
    `aws ssm start-session --target i-xxx`,
]

const options = {
    shell: "zsh",
    encoding: 'utf8',
}

for (const command of commands) {
    console.log(command);
    const result = execSync(command, options as ExecSyncOptions);
    console.log(result);
}
```
It returns the following
```bash
11:44:26.907 session-manager-plugin --version
11:44:26.907 1.2.553.0
11:44:26.911 which session-manager-plugin
11:44:26.911 /usr/local/bin/session-manager-plugin
11:44:26.915 which aws
11:44:26.915 /usr/local/bin/aws
11:44:28.817 aws ssm start-session --target i-xxx
11:44:28.817 Error: Command failed: aws ssm start-session --target i-xxx

SessionManagerPlugin is not found. Please refer to SessionManager Documentation here: http://docs.aws.amazon.com/console/systems-manager/session-manager-plugin-not-found
```

So I'm a bit puzzled here why running aws ssm start-session would fail if session-manager-plugin is successful

I only have this outcome using execSync but I'd really like to understand how it's possible for this to happen

Thanks

**Comment by uldyssian-sh** [comment, marked as answer]
This usually happens because the AWS CLI process cannot find or execute `session-manager-plugin` in the environment it actually runs in (even though `session-manager-plugin --version` works in your interactive shell).

In your case (Node.js `execSync`), the most common cause is that Raycast/Node is running with a different PATH (or different architecture context) than your terminal, so the AWS CLI cannot launch the plugin binary.

First, print the PATH that Node sees and compare it with your terminal:

node -e "console.log(process.env.PATH)"


Then run the AWS CLI with an explicit PATH (adjust the path to where the plugin is installed):

PATH="/usr/local/bin:$PATH" aws ssm start-session --target i-xxx


If you are on macOS with Apple Silicon, also check the architecture mismatch. If AWS CLI is running under one architecture and the plugin under another, the CLI may fail to execute it.

Check architectures:

file /usr/local/bin/aws

file /usr/local/bin/session-manager-plugin


Fix by ensuring both are installed for the same architecture (e.g., both arm64), or reinstall both via the same package manager (Homebrew is typical):

brew reinstall awscli

brew reinstall session-manager-plugin


Finally, confirm what executable the AWS CLI is trying to use from the same environment where it fails:

which aws

which session-manager-plugin

aws --version

session-manager-plugin --version


Once the PATH/architecture is consistent in the Node/Raycast environment, `aws ssm start-session` will find and run the plugin correctly.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
